data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/configuration/ExportConfig.java

```diff
@@ -33,4 +37,15 @@ public Region getAwsRegion() {
         return s3Region != null ? Region.of(s3Region) : null;
     }
 
+    public String getS3SseKmsKeyId() {
+        return s3SseKmsKeyId;
+    }
+
+    @AssertTrue(message = "KMS Key ID must be a valid one.")
+    boolean isKmsKeyIdValid() {
+        // If key id is provided, it should be in a format like
+        // arn:aws:kms:us-west-2:123456789012:key/0a4bc22f-bb96-4ad3-80ca-63b12b3ec147
+        return s3SseKmsKeyId == null || Arn.fromString(s3SseKmsKeyId).resourceAsString() != null;
```

Comment:
I'm not sure we really want this validation to be so strict. Users should be allowed to provide KMS keys as aliases. e.g. `alias/MyAliasedKey`.

Reply:
I don't agree with this. Even from DynamoDB console, there is no way to provide an aliaes, can only provide KMS Key ID.  Note that alias is optional to KMS CMK.  I believe you want more flexibility, but to get Key ID (which is used in the API call) by alias, we will need to have more permissions required (maybe describe keys or something).

Reply:
Also alias may not work for multi-region keys.

Reply:
@daixba , Typically AWS services allow for either specifying an alias or an ARN. So my ask is not to require the user to provide an alias. It is to allow the user to provide either an ARN or an alias.

The [documentation](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_ExportTableToPointInTime.html#DDB-ExportTableToPointInTime-request-S3SseKmsKeyId) on this is unclear, but their restrictions indicate a minimum length of 1 which is compatible with an alias.

We shouldn't follow the Console, but what the API itself allows. If this API does not permit an alias, then I agree this should be only an ARN. But, it appears to support the alias.